clar2wasm/src/words/sequences.rs

```diff
@@ -0,0 +1,1133 @@
+            .binop(BinaryOp::I32Add)
+            .local_set(end_offset);
+
+        // Evaluate the initial value, so that its result is on the data stack
+        generator.traverse_expr(builder, initial)?;
+
+        if seq_len == 0 {
+            // If the sequence is empty, just return the initial value
+            return Ok(());
+        }
+
+        // Define local(s) to hold the intermediate result, and initialize them
+        // with the initial value. Not that we are looping in reverse order, to
+        // pop values from the top of the stack.
+        let mut result_locals = Vec::with_capacity(result_ty.len());
+        for local_ty in result_ty.iter().rev() {
+            let local = generator.module.locals.add(*local_ty);
+            result_locals.push(local);
+            builder.local_set(local);
+        }
+        result_locals.reverse();
+
+        // Define the body of a loop, to loop over the sequence and make the
+        // function call.
+        builder.loop_(loop_body_ty, |loop_| {
+            let loop_id = loop_.id();
+
+            // Load the element from the sequence
+            let elem_size = generator.read_from_memory(loop_, offset, 0, elem_ty);
+
+            // Push the locals to the stack
+            for result_local in result_locals.iter() {
+                loop_.local_get(*result_local);
+            }
+
+            // Call the function
+            loop_.call(generator.func_by_name(func.as_str()));
+
+            // Save the result into the locals (in reverse order as we pop)
+            for result_local in result_locals.iter().rev() {
+                loop_.local_set(*result_local);
+            }
+
+            // Increment the offset by the size of the element, leaving the
+            // offset on the top of the stack
+            loop_
+                .local_get(offset)
+                .i32_const(elem_size)
+                .binop(BinaryOp::I32Add)
+                .local_tee(offset);
+
+            // Loop if we haven't reached the end of the sequence
+            loop_
+                .local_get(end_offset)
+                .binop(BinaryOp::I32LtU)
+                .br_if(loop_id);
+        });
+
+        // Push the locals to the stack
+        for result_local in result_locals.iter() {
+            builder.local_get(*result_local);
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct Append;
+
+impl Word for Append {
+    fn name(&self) -> ClarityName {
+        "append".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut crate::wasm_generator::WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        expr: &SymbolicExpression,
+        args: &[clarity::vm::SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        if args.len() != 2 {
+            return Err(GeneratorError::InternalError(
+                "expected two arguments to 'append'".to_string(),
+            ));
+        }
```

Comment:
I guess I was in a safety mindset at the time. I'll change all accesses of args in this file to use `get_expr` as we discussed in Discord, then remove those checks.

Reply:
Oh, in making that change I also realized that I should not evaluate the arguments out of order. I was doing that in `replace-at?` to get my values on the stack in the right places, but I'll change that so that the expressions are evaluated in order.

Reply:
ok this is solved now.